taxonomy-editor-frontend/src/components/LanguageSelectionDialog.jsx

```diff
@@ -0,0 +1,84 @@
+import {
+  OutlinedInput,
+  InputLabel,
+  MenuItem,
+  FormControl,
+  ListItemText,
+  Select,
+  Checkbox,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Dialog,
+  Button,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import ISO6391 from "iso-639-1";
+
+const LanguageSelectionDialog = ({
+  isDialogOpen,
+  handleClose,
+  mainLanguage,
+  handleLanguagesChange,
+  selectedShownLanguages,
+}) => {
+  const [newShownLanguageCodes, setNewShownLanguageCodes] = useState([]); // Used for storing LCs that are selected in the checkbox (temporary state)
+
+  useEffect(() => {
+    setNewShownLanguageCodes(selectedShownLanguages);
+  }, [selectedShownLanguages]);
+
+  const handleLanguageCheckToggle = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setNewShownLanguageCodes(
+      typeof value === "string" ? value.split(",") : value
+    );
+  };
+
+  return (
+    <Dialog open={isDialogOpen} onClose={handleClose}>
```

Comment:
replace with `<> ...restOfDialogContent</>`
and in ListTranslations use `<Dialog open={isDialogOpen} onClose={handleClose}> <LanguageSelectionDialog .../> </Dialog>` this makes it be mounted and unmounted when open/close